src/H5TSint.c

```diff
@@ -135,26 +135,32 @@ H5TS__init(void)
  *
  * Purpose:  Terminate this interface.
  *
- * Note:     This function is currently registered via atexit() and is called
- *              AFTER H5_term_library().
- *
  * Return:    void
  *
  *-------------------------------------------------------------------------
  */
-void
+int
 H5TS_term_package(void)
 {
     FUNC_ENTER_NOAPI_NOINIT_NOERR
 
+    /* Clean up per-thread library info */
+    H5TS__tinfo_term();
+
+    /* Tell subsequent library opens that thread information is not initialized */
+    H5TS_first_init_s = H5TS_ONCE_INITIALIZER;
+
+    /* Release critical section / mutex for modifying the thread info globals */
+    H5_UNLIKELY(H5TS_mutex_destroy(&H5TS_tinfo_mtx_s));
```

Comment:
Why did you move these destroy+delete out of the H5TS_tinfo_term call?

Reply:
I think this is a holdover from an initial design that split H5TS_term into two routines - I've moved them back into tinfo_term.